react/features/settings/components/web/SettingsButton.js

```diff
@@ -45,11 +50,13 @@ class SettingsButton extends AbstractButton<Props, *> {
      * @returns {void}
      */
     _handleClick() {
-        const { _filmstripOnly, dispatch } = this.props;
+        const { _filmstripOnly, defaultTab, dispatch } = this.props;
 
         sendAnalytics(createToolbarEvent('settings'));
         if (_filmstripOnly) {
             dispatch(openDeviceSelectionPopup());
+        } else if (defaultTab) {
+            dispatch(openSettingsDialog(defaultTab));
         } else {
             dispatch(openSettingsDialog(SETTINGS_TABS.DEVICES));
```

Comment:
Style nit for thought: to avoid an extra elseif clause you could instead do a truthy check fallback in what you pass to `openSettingsDialog` using an || or even setting a default prop to `SETTINGS_TABS.DEVICES` so that defaultTab is always passed in.